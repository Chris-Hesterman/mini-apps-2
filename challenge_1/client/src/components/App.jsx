import React from 'react';
import axios from 'axios';
import ReactPaginate from 'react-paginate';
import styled, { createGlobalStyle } from 'styled-components';
import Title from './Title.jsx';
import Search from './Search.jsx';
import List from './List.jsx';

const GlobalStyle = createGlobalStyle`
  html {
    margin: 0;
    padding: 0;
    height: 100vh;
    width: 100vw;
  }
  body {
    margin: 0;
    padding: 0;
    height: 100%;
    width: 100%
  }
  div#react-paginate {
    height: 100%;
    width: 100%;
    background: tan;
  }
  ul.page-list {
    display: flex;
    list-style-type: none;
    padding: 0;
    width: 400px;
    justify-content: space-around;
    align-items: center;
  }
  li.page-li {
    display: flex;
    justify-content: center;
    align-items: center;
    width: 30px;
    height: 30px;
    border: 2px solid #d1913c;
    border-radius: 25px;
    background: rgba(230, 230, 230, 0.5);
    &:hover {
      background: #d1913c;
      color: white;
      cursor: pointer;
    }
  }
  li.next, li.previous {
    display: flex;
    justify-content: center;
    align-items: center;
    border: 2px solid #d1913c;
    border-radius: 25px;
    padding: 0 10px;
    height: 25px;
    color: black;
    &:hover {
      background: green;
      color: white;
      cursor: pointer;
      text-decoration: underline;
    }
  }
  li.selected {
    border-color: lime;
  }
  a.page-link, a.next-link, a.prev-link {
    outline: none;
  }
`;

const StyledWrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: flex-start;
  font-family: sans-serif;
  height: 100%;
  width: 100%;
  background: tan;
`;

const StyledDiv = styled.div`
  position: absolute;
  bottom: 0;
  margin-bottom: 20px;
`;

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      events: [],
      pageCount: 10,
      currentPage: 1,
      currentQuery: ''
    };
    this.getEventPage = this.getEventPage.bind(this);
    this.handlePage = this.handlePage.bind(this);
  }

  getEventPage(query, page = this.state.currentPage) {
    console.log(query);
    const getURL = `http://127.0.0.1:3000/events/?q=${query}&_sort=date&_page=${page}&_limit=4`;
    console.log('get');
    axios
      .get(getURL)
      .then((response) => {
        console.log(response);
        this.setState({
          events: response.data,
          currentQuery: query
        });
      })
      .catch((err) => {
        console.log(error);
      });
  }

  handlePage(data) {
    console.log(data.selected);
    this.setState((prevState, currState) => {
      return { currentPage: data.selected + 1 };
    }, this.getEventPage(this.state.currentQuery, data.selected + 1));
  }

  render() {
    return (
      <StyledWrapper>
        <GlobalStyle />
        <Title />
        <Search send={this.getEventPage} />
        <List events={this.state.events} />
        <StyledDiv>
          <ReactPaginate
            pageCount={this.state.pageCount}
            pageRangeDisplay={this.state.pageCount}
            marginPagesDisplayed={1}
            onPageChange={this.handlePage}
            activeClassName={'selected'}
            previousClassName={'previous'}
            nextClassName={'next'}
            containerClassName={'page-list'}
            pageClassName={'page-li'}
            previousLinkClassName={'prev-link'}
            nextLinkClassName={'next-link'}
            pageLinkClassName={'page-link'}
            activeLinkClassName={'active-link'}
          />
        </StyledDiv>
      </StyledWrapper>
    );
  }
}

export default App;
