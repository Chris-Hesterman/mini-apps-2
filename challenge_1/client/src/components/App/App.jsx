import React from 'react';
import axios from 'axios';
import styled from 'styled-components';
import ReactPaginate from 'react-paginate';
import Title from '../Title.jsx';
import Search from '../Search/Search.jsx';
import List from '../List/List.jsx';
import Edit from '../Edit.jsx';
import { GlobalStyle, StyledAppWrapper, StyledAppDiv } from './App.css.js';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      events: [],
      pageCount: 3,
      currentPage: 1,
      currentQuery: '',
      edit: false
    };
    this.getEventPage = this.getEventPage.bind(this);
    this.handlePage = this.handlePage.bind(this);
    this.handleEdit = this.handleEdit.bind(this);
    this.saveEdit = this.saveEdit.bind(this);
    this.putEvents = this.putEvents.bind(this);
  }

  getEventPage(query, page = 1) {
    const getURL = `/events?q=${query}&page=${page}&limit=4`;
    console.log('hitting get request');
    axios
      .get(getURL)
      .then((response) => {
        let pageCount = response.data.totalPages;
        console.log('axios', response);
        this.setState({
          events: response.data.docs,
          currentQuery: query,
          currentPage: page,
          pageCount: pageCount
        });
      })
      .catch((err) => {
        console.log(err);
      });
  }

  putEvents(data) {
    this.setState({ events: data, edit: false });
    console.log('Made it to put', data);

    axios
      .put(`/events`, data)
      .then((response) => {
        console.log('events updated!:', response.config.data);
      })
      .catch((err) => {
        console.log('Sorry, did not get saved. Please try again');
      });
  }

  handlePage(data) {
    if (this.state.events.length) {
      this.setState((prevState, currState) => {
        return { currentPage: data.selected + 1 };
      }, this.getEventPage(this.state.currentQuery, data.selected + 1));
    }
  }

  handleEdit(e) {
    this.setState((prev, curr) => {
      return { edit: !this.state.edit };
    });
  }

  saveEdit(data) {
    this.putEvents(data);
  }

  render() {
    const eventsCopy = this.state.events.map((event) => {
      return Object.assign({}, event);
    });

    return (
      <StyledAppWrapper>
        <GlobalStyle />
        <Title />
        <Search send={this.getEventPage} />
        {this.state.edit ? (
          <Edit
            done={this.handleEdit}
            eventData={eventsCopy}
            save={this.saveEdit}
          />
        ) : (
          <List
            eventData={eventsCopy}
            currentQuery={this.state.currentQuery}
            edit={this.handleEdit}
          />
        )}

        <StyledAppDiv>
          <ReactPaginate
            pageCount={this.state.pageCount}
            pageRangeDisplay={this.state.pageCount}
            marginPagesDisplayed={1}
            onPageChange={this.handlePage}
            activeClassName={'selected'}
            previousClassName={'previous'}
            nextClassName={'next'}
            containerClassName={'page-list'}
            pageClassName={'page-li'}
            previousLinkClassName={'prev-link'}
            nextLinkClassName={'next-link'}
            pageLinkClassName={'page-link'}
            activeLinkClassName={'active-link'}
            forcePage={this.state.currentPage - 1}
          />
        </StyledAppDiv>
      </StyledAppWrapper>
    );
  }
}

export default App;
