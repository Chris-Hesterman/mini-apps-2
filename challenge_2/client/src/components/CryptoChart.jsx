import React from 'react';
import styled from 'styled-components';
import Chart from 'chart.js';
import { StyledChartWrapper, StyledP } from '../styles.js';
import PropTypes from 'prop-types';

class CryptoChart extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
    this.canvasRef = React.createRef();
    this.makeChart = this.makeChart.bind(this);
  }

  makeChart() {
    const ctx = this.canvasRef.current;
    const labels = Object.keys(this.props.bpi);
    const data = Object.values(this.props.bpi);

    Chart.defaults.global.elements.point.borderColor = 'black';
    Chart.defaults.global.elements.point.borderWidth = 1;
    Chart.defaults.global.elements.point.pointStyle = 'circle';
    Chart.defaults.global.elements.point.radius = 0;
    Chart.defaults.global.elements.point.hitRadius = 8;

    const myChart = new Chart(ctx, {
      type: 'line',
      data: {
        labels: labels,
        datasets: [
          {
            label: 'Bitcoin Value, USD',
            data: data,
            backgroundColor: ['rgba(255, 99, 132, 0.3)'],
            borderColor: ['rgba(255, 99, 132, 1)'],
            borderWidth: 2,
            lineTension: 0
          }
        ]
      },
      options: {
        scales: {
          yAxes: [
            {
              gridLines: {
                color: 'rgba(0, 0, 0, .2)'
              },
              ticks: {
                beginAtZero: true,
                fontColor: '#222'
              }
            }
          ],
          xAxes: [
            {
              gridLines: {
                color: 'rgba(0, 0, 0, .2)'
              },
              ticks: {
                fontColor: '#222'
              },
              autoSkip: true,
              autoSkipPadding: 25
            }
          ]
        }
      }
    });
  }

  componentDidMount() {
    this.makeChart();
  }

  componentDidUpdate(prevProps) {
    if (this.props !== prevProps) {
      this.makeChart();
    }
  }

  render() {
    return (
      <StyledChartWrapper>
        <canvas id="chart" ref={this.canvasRef}></canvas>
        <StyledP>{this.props.dis}</StyledP>
      </StyledChartWrapper>
    );
  }
}

CryptoChart.propTypes = {
  bpi: PropTypes.object,
  dis: PropTypes.string
};

export default CryptoChart;
