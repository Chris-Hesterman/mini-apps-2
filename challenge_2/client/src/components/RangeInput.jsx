import React from 'react';
import styled from 'styled-components';
import moment from 'moment';
import { StyledForm, StyledInput, StyledButton } from '../styles.js';
import PropTypes from 'prop-types';

class RangeInput extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      start: '',
      end: ''
    };
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleChange(e) {
    const endOrStart = e.target.classList[e.target.classList.length - 1];
    console.log(endOrStart);

    this.setState({ [endOrStart]: e.currentTarget.value });
  }

  handleSubmit(e) {
    e.preventDefault();
    const start = moment(this.state.start).format('YYYY-MM-DD');
    const end = moment(this.state.end).format('YYYY-MM-DD');

    this.props.getRange(start, end);
  }

  render() {
    return (
      <StyledForm onSubmit={this.handleSubmit}>
        <StyledInput
          value={this.state.start}
          type="text"
          className="start"
          placeholder="Enter start date after 7/17/2010"
          onChange={this.handleChange}
          autoFocus
        ></StyledInput>
        <StyledInput
          value={this.state.end}
          type="text"
          className="end"
          placeholder="Enter end date"
          onChange={this.handleChange}
        ></StyledInput>
        <StyledButton type="submit">Submit</StyledButton>
      </StyledForm>
    );
  }
}

RangeInput.propTypes = {
  getRange: PropTypes.func
};

export default RangeInput;
