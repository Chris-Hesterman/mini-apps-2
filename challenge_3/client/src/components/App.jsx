import React from 'react';
import Pinpad from './Pinpad.jsx';
import GameScore from './GameScore.jsx';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      currentFrame: 1,
      1: [],
      2: [],
      3: [],
      4: [],
      5: [],
      6: [],
      7: [],
      8: [],
      9: [],
      10: []
    };
    this.addPins = this.addPins.bind(this);
    this.advanceFrame = this.advanceFrame.bind(this);
    this.checkFrameLength = this.checkFrameLength.bind(this);
  }

  checkFrameLength() {
    if (
      this.state[this.state.currentFrame.toString()].length === 2 &&
      this.state.frameNumber !== 10
    ) {
      this.advanceFrame();
    }
  }

  addPins(numToppledPins) {
    console.log(numToppledPins);
    const frame = [...this.state[this.state.currentFrame.toString()]];

    if (!frame.length) {
      if (numToppledPins === 10) {
        frame.push('X');
        console.log(frame);
        this.setState({ frame });

        return;
      }
    }

    if (frame[0] + numToppledPins === 10) {
      frame.push('/');
      console.log(frame);
      this.setState({ frame });

      return;
    }

    if (frame.length < 2) {
      frame.push(numToppledPins);
      console.log(frame);
      this.setState({ [this.state.currentFrame.toString()]: frame });

      return;
    }
  }

  advanceFrame() {
    this.setState((prevState) => {
      return { currentFrame: prevState.currentFrame + 1 };
    });
  }

  componentDidUpdate(prevProps, prevState) {
    if (
      this.state[this.state.currentFrame.toString()].length !==
      prevState[this.state.currentFrame.toString()].length
    ) {
      this.checkFrameLength();
    }
  }

  render() {
    return (
      <div>
        <Pinpad throwBall={this.addPins} />
        <GameScore frames={this.state} />
      </div>
    );
  }
}

export default App;
