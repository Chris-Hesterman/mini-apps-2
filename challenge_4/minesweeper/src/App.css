.App {
  text-align: center;
  width: 800px;
  height: 800px;
  color: greenyellow;
  text-shadow: 1px 1px black, -1px 1px black, -1px -1px black, 1px -1px black;
  letter-spacing: 3px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

h1 {
  margin-top: 0;
}
