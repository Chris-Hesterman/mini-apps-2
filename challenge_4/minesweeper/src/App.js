import React from 'react';
import './App.css';
import Board from './features/board/Board.js';

function App() {
  return (
    <div className="App">
      <h1>ugly game</h1>
      <Board />
    </div>
  );
}

export default App;
