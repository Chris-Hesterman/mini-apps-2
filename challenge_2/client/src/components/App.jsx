import React from 'react';
import styled, { createGlobalStyle } from 'styled-components';
import CryptoChart from './CryptoChart.jsx';

const GlobalStyle = createGlobalStyle`
  html {
    height: 100vh;
    padding: 0;
    margin: 0;
  }
  body {
    height: 100%;
    padding: 0;
    margin: 0;
  }
  div#app {
    width: 100%;
    height: 100%;
    display: flex;
    justify-content: center;
  }
`;

const StyledWrapper = styled.div`
  width: 80%;
  height: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
`;
class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    return (
      <StyledWrapper>
        <GlobalStyle />
        <h1>Yes, it's a nifty chart!</h1>
        <CryptoChart />
      </StyledWrapper>
    );
  }
}

export default App;
