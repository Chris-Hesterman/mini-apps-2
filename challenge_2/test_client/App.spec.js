import React from 'react';
import PropTypes from 'prop-types';
import checkPropTypes from 'check-prop-types';
import Enzyme from 'enzyme';
import { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import mockAxios from 'axios';
import Chart from 'chart.js';
import canvas from 'canvas';
import App from '../client/src/components/App.jsx';
import CryptoChart from '../client/src/components/CryptoChart.jsx';
import RangeInput from '../client/src/components/RangeInput.jsx';

Enzyme.configure({ adapter: new Adapter() });

const response = {
  data: {
    bpi: {
      '2018-02-22': 9830.4263,
      '2018-02-23': 10149.4625,
      '2018-02-24': 9682.3825
    },
    dis:
      'This data was produced from the CoinDesk Bitcoin Price Index. BPI value data returned as USD.'
  }
};

describe('App', () => {
  const wrapper = shallow(<App />);

  it('Should render correctly', () => {
    expect(wrapper.find('.app').length).toBe(1);
  });

  it('Should render one RangeInput component', () => {
    expect(wrapper.find('.input').length).toBe(1);
  });

  it('Fetches data from server, using default range when none provided', () => {
    const wrapper = mount(<App />);

    mockAxios.get.mockImplementationOnce(() => {
      Promise.resolve(response);
    });

    expect(mockAxios.get).toHaveBeenCalledTimes(2);
    expect(mockAxios.get).toHaveBeenCalledWith(
      'http://127.0.0.1:3000/crypto?start=2020-03-01&end=2020-04-22'
    );
  });

  it('renders one and only one CryptoChart component only when data is received', () => {
    expect(wrapper.find('.chart').length).toBe(0);

    wrapper.setState({ data: response.data });

    expect(wrapper.find('.chart').length).toBe(1);
    expect(wrapper.find('.chart').length).toBeLessThan(2);
  });
});
