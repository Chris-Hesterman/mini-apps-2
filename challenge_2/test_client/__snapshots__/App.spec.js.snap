// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
"<styles__StyledAppWrapper className=\\"app\\">
  <Memo(GlobalStyleComponent) />
  <styles__StyledH1>
    Check Historic Bitcoin Value
  </styles__StyledH1>
  <RangeInput getRange={[Function: bound fetchData]} className=\\"input\\" />
</styles__StyledAppWrapper>"
`;

exports[`App renders one and only one CryptoChart component only when data is received 1`] = `
"<styles__StyledAppWrapper className=\\"app\\">
  <Memo(GlobalStyleComponent) />
  <styles__StyledH1>
    Check Historic Bitcoin Value
  </styles__StyledH1>
  <RangeInput getRange={[Function: bound fetchData]} className=\\"input\\" />
  <CryptoChart bpi={{ '2018-02-22': 9830.4263, '2018-02-23': 10149.4625, '2018-02-24': 9682.3825 }} dis={[undefined]} className=\\"chart\\" />
</styles__StyledAppWrapper>"
`;
